js/packages/teams-ai/src/MessageExtensions.ts

```diff
@@ -351,50 +360,41 @@ export class MessageExtensions<TState extends TurnState> {
 
     /**
      * Registers a handler that implements a Link Unfurling based Message Extension.
-     * @param {(string | RegExp | RouteSelector | string[] | RegExp[] | RouteSelector[])} commandId - ID of the command(s) to register the handler for.
      * @param {(context: TurnContext, state: TState, url: string) => Promise<MessagingExtensionResult>} handler - Function to call when the command is received.
      * @param {TurnContext} handler.context - Context for the current turn of conversation with the user.
      * @param {TState} handler.state - Current state of the turn.
      * @param {string} handler.url - The URL that should be unfurled.
      * @returns {Application<TState>} The application for chaining purposes.
      */
     public queryLink(
```

Comment:
I came across this issue when working on Auth. Even though queryLink doesn't filter by command, it can be filtered by the incoming url. I have made those changes as part of my Auth pr: https://github.com/microsoft/teams-ai/pull/670/files#diff-11a6f41116a773eaa5cbad4f5175a802d703e38ed8a809160d73ca7766975ba3. Here's how end users can use this method:
```ts
// Matches any link that ends in `.com`
app.messageExtensions.queryLink(/.+\.com/, async (context: TurnContext, state: DefaultTurnState, url: string) => {
    // return an attachment
})
```
 Thoughts?

Reply:
The regex to filter for a url would be much much longer, needing to account for all the possible domain suffixes, potentially check for http/https, etc.

I'm not opposed to the change. But according to schema Should continue discussion on: https://github.com/microsoft/teams-ai/issues/678 Filtering is not necessary and can be handled within the handler.

Reply:
This is just an example. I see this filter being useful when only a specific domain should be captured like `https://www.youtube.com`. And the regex filter will come into play to handle variations like `http/s` or `.ca/.com.org`....etc